iothub/device/src/ClientOptions.cs

```diff
@@ -13,5 +13,13 @@ public class ClientOptions
         /// This feature is currently supported only over MQTT.
         /// </summary>
         public string ModelId { get; set; }
+
+        /// <summary>
+        /// The transport settings to use for all file upload operations, regardless of what protocol the device
+        /// client is configured with. All file upload operations take place over https. If no client certificates
+        /// are configured to this object, then file upload operations will use the client certificates configured
```

Comment:
Why not make it mandatory to provide client certificate for file upload in `FileUploadTransportSettings`?
Also, what was the restriction in accepting an HttpClient directly?

Reply:
Because it isn't mandatory to provide FileUploadTransportSettings at all, much less a field on them

Reply:
I investigated taking the http client directly, but there is http client handler class that we could take that exists in all the frameworks that we support. Plus there are just layers and layers of HTTP code underneath all of this, and its a lot safer to just re-use that logic than it is to start dependency injecting clients into that mess

Reply:
hmm, I don't mean to be a pain about this, but since this operation doesn't really have any of the features that we get from the delegating handlers pipeline, couldn't we forgo all of that and directly make the http call within the internal client (with the supplied http client)?